integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/partition/CarbonAlterTableAddHivePartitionCommand.scala

```diff
@@ -153,6 +153,23 @@ case class CarbonAlterTableAddHivePartitionCommand(
         CarbonLoaderUtil.addDataIndexSizeIntoMetaEntry(newMetaEntry, loadModel.getSegmentId, table)
         // Make the load as success in table status
         CarbonLoaderUtil.recordNewLoadMetadata(newMetaEntry, loadModel, false, false)
+
+        // Normally, application will use Carbon SDK to write files into a partition folder, then
+        // add the folder to partitioned carbon table.
+        // If there are many threads writes to the same partition folder, there will be many
+        // carbon index files, and it is not good for query performance since all index files
+        // need to be read to spark driver.
+        // So, here trigger to merge the index files by sending an event
+        val alterTableModel = AlterTableModel(
```

Comment:
@jackylk : Everytime when we do alter table add partition, This will trigger merge-index task for all the segments.  May be we can trigger for only new partition (segment) ?

Reply:
For normal minor compaction, is it triggering index merge for all segments?
One solution is to add one parameter in AlterTableMergeIndexEvent to specify the segmentId to do merge index. What do you think?

Reply:
yeah, like `custom compaction`, we can specify the segment id for which it need to create merge index

Reply:
`AlterTableModel()` already have an argument `customSegmentIds`, we can check it can work by passing only current partition's segment id.